packages/loader/container-utils/src/error.ts

```diff
@@ -85,9 +110,38 @@ export class DataProcessingError extends LoggingError implements IErrorBase {
     constructor(errorMessage: string, props?: ITelemetryProperties) {
         super(errorMessage, props);
     }
+
+    /**
+     * Conditionally coerce the throwable input into a DataProcessingError.
+     * @param error - Throwable input to be converted.
+     * @param message - Sequenced message (op) to include info about via telemetry props
+     * @returns Either a new DataProcessingError, or (if wrapping is deemed unnecessary) the given error
+     */
+    static wrapIfUnrecognized(
+        error: any,
+        message: ISequencedDocumentMessage | undefined,
+    ): ICriticalContainerError {
+        const messagePropsToLog = message !== undefined
+            ? extractSafePropertiesFromMessage(message)
+            : undefined;
+
+        // Don't coerce if it's already a recognized LoggingError
+        if (isValidLoggingError(error)) {
+            if (messagePropsToLog !== undefined) {
+                error.addTelemetryProperties(messagePropsToLog);
+            }
+            return error;
+        }
+
+        const newErrorFn =
+            (errMsg: string, props?: ITelemetryProperties) =>
+                new DataProcessingError(errMsg, { ...props, ...messagePropsToLog });
+
+        return wrapError(error, newErrorFn);
```

Comment:
I think the following flow is a bit more readable and ensures that properties always applied in the same order in both branches:

```typescript
const error2 = isValidLoggingError(error) ? error : wrapError(error, DataProcessingError);
error2.addTelemetryProperties(messagePropsToLog);
return error2;
```

Assumes 
```
export function wrapError<T extends LoggingError>(
    error: any,
    newErrorFn: new (m: string, p?: ITelemetryProperties) => T,
```

i.e. that ctor is passed in for an object.

Reply:
Similar feedback for other places (like CreateContainerError)

Reply:
Great feedback.  I can't do the `newErrorFn: new`... part because sometimes you need a more interesting function.  But otherwise, did it.